ext/ITensorNetworksEinExprsExt/src/ITensorNetworksEinExprsExt.jl

```diff
@@ -0,0 +1,58 @@
+module ITensorNetworksEinExprsExt
+
+using ITensors: Index, ITensor, Algorithm
+using ITensorNetworks:
+  ITensorNetworks,
+  ITensorNetwork,
+  vertextype,
+  vertex_data,
+  externalinds,
+  contraction_sequence
+using EinExprs: EinExprs, EinExpr, einexpr, SizedEinExpr
+
+function to_einexpr(tn::ITensorNetwork)
+  IndexType = Any
+  VertexType = vertextype(tn)
+
+  tensor_exprs = EinExpr{IndexType}[]
+  tensor_inds_to_vertex = Dict{Set{IndexType},VertexType}()
+  inds_dims = Dict{IndexType,Int}()
+
+  for v in vertices(tn)
+    tensor_v = tn[v]
+    inds_v = collect(inds(tensor_v))
+    push!(tensor_exprs, EinExpr{IndexType}(; head=inds_v))
+    tensor_inds_to_vertex[Set(inds_v)] = key
+    merge!(inds_dims, Dict(inds_v .=> size(tensor_v)))
+  end
+
+  externalinds_tn = collect(externalinds(tn))
+  expr = SizedEinExpr(sum(tensor_exprs; skip=externalinds_tn), inds_dims)
+
+  return expr, tensor_inds_to_vertex
+end
+
+function EinExprs.einexpr(tn::ITensorNetwork; optimizer::EinExprs.Optimizer)
+  expr, _ = to_einexpr(tn)
+  return einexpr(optimizer, expr)
+end
+
+function ITensorNetworks.contraction_sequence(
+  alg::Algorithm"einexpr", tn::Vector{ITensor}; kwargs...
+)
+  return contraction_sequence(alg, ITensorNetwork(tn); kwargs...)
+end
```

Comment:
It seems like this should already be defined generically for any `Algorithm` input, is that not the case? If not this should be added to the main package.

Reply:
I think that is not the case. What I've seen is that `contraction_sequence` is called with a `Vector{ITensor}` by default.

Reply:
Perhaps we should make this in line with the other functions in `contraction_sequences.jl` to resolve it. Specifically, we currently have the logic:
```
function contraction_sequence(tn::AbstractITensorNetwork; kwargs...)
  seq_linear_index = contraction_sequence(Vector{ITensor}(tn); kwargs...)
  # TODO: Use Functors.fmap?
  return deepmap(n -> Key(vertices(tn)[n]), seq_linear_index)
end
```
So here really we should have `to_einexpr(tn::ITensorNetwork)` and `contraction_sequence(
  ::Algorithm"einexpr", tn::ITensorNetwork{T}; kwargs..)` take a `Vector{ITensor}` as the argument instead? 

I think the logic of `to_einexpr` would be almost unchanged, except the call `externalinds(tn)` should be replaced by `noncommondinds(ts)` where `ts::Vector{ITensor}`?

Reply:
Ah I understand. This is the standard logic for other `contraction_sequence` backends:
https://github.com/mtfishman/ITensorNetworks.jl/blob/3fc272e7ff7bc1650b76695d2bf883ef7ef7d618/src/contraction_sequences.jl#L5-L9
i.e. it converts from `ITensorNetwork` to `Vector{ITensor}`, gets a sequence in terms of integer vertex labels, then converts that sequence back to the vertex labels/names of the `ITensorNetwork` (wrapped in the `Key` type to disambiguate the case where the vertices themselves are of type `Vector`).

It seems like we could in principle do that here, though the current code makes use of some `ITensorNetwork` functions like `externalinds`. We could just keep that in mind for the future.

Reply:
Just saw your post @JoeyT1994. Seems reasonable to rewrite these functions in terms of `Vector{ITensor}` instead in this PR.

Reply:
should i refactor it to work with `Vector{ITensor}` or is it ok if i just transform to an `ITensorNetwork`?

Reply:
I just made suggestions on `to_einexpr()` for how to change it to take `Vector{ITensor}` instead. The same logic can be done with `tensor_inds_to_vertex` and then you should be able to get rid of the `contraction_sequence` definitions in this file.